File: augur-engine/src/collect/tavily.py
"""Tavily news search API collector."""

from __future__ import annotations

import os
from datetime import datetime, timezone

import httpx

from ..config.types import Signal

TAVILY_API = "https://api.tavily.com/search"


async def collect_tavily(query: str) -> Signal:
    api_key = os.environ.get("TAVILY_API_KEY")
    if not api_key:
        raise RuntimeError("TAVILY_API_KEY not set")

    async with httpx.AsyncClient(timeout=30) as client:
        resp = await client.post(TAVILY_API, json={
            "api_key": api_key,
            "query": query,
            "search_depth": "advanced",
            "max_results": 10,
            "include_answer": False,
            "include_raw_content": False,
        })
        resp.raise_for_status()
        data = resp.json()

    return Signal(
        source="tavily",
        fetched_at=datetime.now(timezone.utc).isoformat(),
        query=query,
        content=[{
            "title": r.get("title", ""),
            "url": r.get("url", ""),
            "snippet": r.get("content", ""),
            "score": r.get("score", 0),
            "date": r.get("published_date"),
        } for r in data.get("results", [])],
    )

File: augur-engine/src/collect/tavily.ts
import type { Signal } from '../config/types.js';

const TAVILY_API = 'https://api.tavily.com/search';

interface TavilyResult {
  title: string;
  url: string;
  content: string;
  score: number;
  published_date?: string;
}

interface TavilyResponse {
  results: TavilyResult[];
  query: string;
}

/** Collect news signals via Tavily search API */
export async function collectTavily(query: string): Promise<Signal> {
  const apiKey = process.env['TAVILY_API_KEY'];
  if (!apiKey) throw new Error('TAVILY_API_KEY not set');

  const res = await fetch(TAVILY_API, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      api_key: apiKey,
      query,
      search_depth: 'advanced',
      max_results: 10,
      include_answer: false,
      include_raw_content: false,
    }),
  });

  if (!res.ok) {
    throw new Error(`Tavily API error: ${res.status} ${res.statusText}`);
  }

  const data = (await res.json()) as TavilyResponse;

  return {
    source: 'tavily',
    fetchedAt: new Date().toISOString(),
    query,
    content: data.results.map((r) => ({
      title: r.title,
      url: r.url,
      snippet: r.content,
      score: r.score,
      date: r.published_date,
    })),
  };
}
